laurat007/assignment3/Word.java

```diff
@@ -41,68 +40,33 @@ static private void backtrack(int startingCellX, int startingCellY, Grid G,
         // stop condition
         if(! G.dictionary.isPrefix(word)) {
+            visited[startingCellX][startingCellY] = false;
             return;
         }
         // if solution found, will be added
         if(G.dictionary.isWord(word))
             solution.add(word);
         
         // try moving
-        // left up
-        if(G.isValid(startingCellX - 1, startingCellY - 1, visited)) {
-            backtrack(startingCellX - 1, startingCellY - 1, G, solution, 
-                word + G.grid[startingCellX - 1][startingCellY - 1], visited);
-            visited[startingCellX - 1][startingCellY - 1] = false;
-        }
-        
-        // up
-        if(G.isValid(startingCellX - 1, startingCellY, visited)) {
-            backtrack(startingCellX - 1, startingCellY, G, solution, 
-                word + G.grid[startingCellX - 1][startingCellY], visited);
-            visited[startingCellX - 1][startingCellY] = false;
-        }
-        
-        // right up
-        if(G.isValid(startingCellX - 1, startingCellY + 1, visited)) {
-            backtrack(startingCellX - 1, startingCellY + 1, G, solution, 
-                word + G.grid[startingCellX - 1][startingCellY + 1], visited);
-            visited[startingCellX - 1][startingCellY + 1] = false;
-        }
-        
-        // left
-        if(G.isValid(startingCellX, startingCellY - 1, visited)) {
-            backtrack(startingCellX, startingCellY - 1, G, solution, 
-                word + G.grid[startingCellX][startingCellY - 1], visited);
-            visited[startingCellX][startingCellY - 1] = false;
-        }
-        
-        // right
-        if(G.isValid(startingCellX, startingCellY + 1, visited)) {
-            backtrack(startingCellX, startingCellY + 1, G, solution, 
-                word + G.grid[startingCellX][startingCellY + 1], visited);
-            visited[startingCellX][startingCellY + 1] = false;
-        }
-        
-        // left down
-        if(G.isValid(startingCellX + 1, startingCellY - 1, visited)) {
-            backtrack(startingCellX + 1, startingCellY - 1, G, solution, 
-                word + G.grid[startingCellX + 1][startingCellY - 1], visited);
-            visited[startingCellX + 1][startingCellY - 1] = false;
-        }
-        
-        // down
-        if(G.isValid(startingCellX + 1, startingCellY, visited)) {
-            backtrack(startingCellX + 1, startingCellY, G, solution, 
-                word + G.grid[startingCellX + 1][startingCellY], visited);
-            visited[startingCellX + 1][startingCellY] = false;
-        }
-        
-        // right down
-        if(G.isValid(startingCellX + 1, startingCellY + 1, visited)) {
-            backtrack(startingCellX + 1, startingCellY + 1, G, solution, 
-                word + G.grid[startingCellX + 1][startingCellY + 1], visited);
-            visited[startingCellX + 1][startingCellY + 1] = false;
-        }
+        int[] deltaX = {startingCellX + 1, startingCellX, startingCellX - 1};
```

Comment:
It is better, it's more or less what I meant but you are making it more complicated than it has to be.
First of all: you are initiating those arrays of deltas with every call to backtrack (which you re calling recursively). Instead you could have one static array of deltas (the deltas never change: -1, 0, 1) and apply them to different values of x and y. 
Secondly you can do this in one go inside of a loop, instead of making precomputation and then looping over it.